common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationResultFactory.java

```diff
@@ -0,0 +1,95 @@
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.util.HashMap;
+
+public class MicrosoftStsAuthorizationResultFactory extends AuthorizationResultFactory {
+
+    private static final String TAG = MicrosoftStsAuthorizationResultFactory.class.getSimpleName();
+
+    public static final String MSSTS_AUTHORIZATION_FINAL_URL = "com.microsoft.identity.client.finalUrl";
+
+    @Override
+    public AuthorizationResult createAuthorizationResult(final int resultCode, final Intent data) {
+        if (data == null) {
+            return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                    MicrosoftAuthorizationErrorResponse.AUTHORIZATION_FAILED, MicrosoftAuthorizationErrorResponse.NULL_INTENT);
+        }
+        switch (resultCode) {
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL:
+                Logger.verbose(TAG, null, "User cancel the request in webview.");
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.USER_CANCEL,
+                        MicrosoftAuthorizationErrorResponse.USER_CANCEL, MicrosoftAuthorizationErrorResponse.USER_CANCELLED_FLOW);
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_COMPLETE:
+                final String url = data.getStringExtra(MSSTS_AUTHORIZATION_FINAL_URL);
+                return parseUrlAndCreateAuthorizationResponse(url);
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR:
+                // This is purely client side error, possible return could be chrome_not_installed or the request intent is
+                // not resolvable
+                final String error = data.getStringExtra(ERROR_CODE);
+                final String errorDescription = data.getStringExtra(ERROR_DESCRIPTION);
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL, error, errorDescription);
+
+            default:
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                        MicrosoftAuthorizationErrorResponse.UNKNOWN_ERROR, MicrosoftAuthorizationErrorResponse.UNKNOWN_RESULT_CODE);
+        }
+    }
+
+    private MicrosoftStsAuthorizationResult createAuthorizationResultWithErrorResponse(final AuthorizationStatus authStatus,
+                                                                                       final String error,
+                                                                                       final String errorDescription) {
+        Logger.info(TAG, null, "Error is returned from webview redirect");
```

Comment:
Consider using two-arg log statements if correlationId is null

Reply:
Done.